pyloidal/cocos.py

```diff
@@ -0,0 +1,221 @@
+        raise ValueError(f"sigma_rpz should be either 1 or -1, found {sigma_rpz}")
+    if sigma_rtp != 1 and sigma_rtp != -1:
+        raise ValueError(f"sigma_rtp should be either 1 or -1, found {sigma_rtp}")
+
+    sigma_to_cocos_dict = {
+        (+1, +1, +1): 1,  # +Bp, +rpz, +rtp
+        (+1, -1, +1): 2,  # +Bp, -rpz, +rtp
+        (-1, +1, -1): 3,  # -Bp, +rpz, -rtp
+        (-1, -1, -1): 4,  # -Bp, -rpz, -rtp
+        (+1, +1, -1): 5,  # +Bp, +rpz, -rtp
+        (+1, -1, -1): 6,  # +Bp, -rpz, -rtp
+        (-1, +1, +1): 7,  # -Bp, +rpz, +rtp
+        (-1, -1, +1): 8,  # -Bp, -rpz, +rtp
+    }
+    result = sigma_to_cocos_dict[(sigma_bp, sigma_rpz, sigma_rtp)]
+    return result if psi_by_2pi else result + 10
+
+
+def identify_cocos(
+    b_toroidal: float,
+    plasma_current: float,
+    safety_factor: ArrayLike,
+    poloidal_flux: ArrayLike,
+    clockwise_phi: Optional[bool] = None,
+    minor_radii: Optional[ArrayLike] = None,
+) -> Tuple[int, ...]:
+    r"""
+    Determine which COCOS coordinate system is in use. Returns all possible conventions.
+
+    Parameters
+    ----------
+    b_toroidal
+        Toroidal magnetic field, with sign. Should be in units of Tesla.
+    plasma_current
+        Plasma current, with sign. Should be in units of Amperes.
+    safety_factor
+        Safety factor profile, with sign, as a function of ``poloidal_flux``. Usually
+        denoted :math:`q`. Should be defined with the first element on the magnetic
+        axis, and subsequent elements on successively larger flux surfaces.
+    poloidal_flux
+        The profile of the poloidal flux function, with sign. Usually denoted
+        :math:`\psi`. Should be defined with the first element on the magnetic
+        axis, and subsequent elements on successively larger flux surfaces.
+    clockwise_phi
+        When viewing tokamak coordinates from above, does the toroidal angle
+        :math:`\phi` increase in the clockwise direction? This is required to identify
+        odd vs even COCOS. This cannot be determined from the output of a code alone.
+        An easy way to determine this is to answer the question: is positive
+        :math:`B_\text{toroidal}` clockwise?
+    minor_radii
+        Th minor radius of each flux surface as function of ``poloidal_flux``. This is
+        required to identify whether :math:`\psi` contains a factor of :math:`2\pi`.
+
+    Returns
+    -------
+    Tuple[int, ...]
+        All possible conventions are returned. If both optional arguments are provided,
+        the returned tuple should have length 1.
+
+    """
+
+    if clockwise_phi is None:
+        return tuple(
+            itertools.chain.from_iterable(
+                identify_cocos(
+                    b_toroidal,
+                    plasma_current,
+                    safety_factor,
+                    poloidal_flux,
+                    x,
+                    minor_radii,
+                )
+                for x in (False, False)
```

Comment:
Should be?

```suggestion
                for x in (True, False)
```

Reply:
Yup, thanks for catching that. Ideally the tests should have caught it, but the ones I adapted from OMAS didn't test multiple returns.